e2etests/tests/advertisement.go

```diff
@@ -284,6 +284,62 @@ var _ = ginkgo.Describe("Advertisement", func() {
 					}
 				},
 			}),
+			ginkgo.Entry("IPV4 - Advertise with communities", params{
+				ipFamily: ipfamily.IPv4,
+				vrf:      "",
+				myAsn:    infra.FRRK8sASN,
+				prefixes: []string{"192.168.0.0/24", "192.168.1.0/24"},
+				modifyPeers: func(ppV4 []config.Peer, ppV6 []config.Peer) {
+					for i := range ppV4 {
+						ppV4[i].Neigh.ToAdvertise.Allowed.Mode = frrk8sv1beta1.AllowAll
+						ppV4[i].Neigh.ToAdvertise.PrefixesWithCommunity = []frrk8sv1beta1.CommunityPrefixes{
+							{
+								Community: "10:100",
+								Prefixes:  []string{"192.168.0.0/24"},
+							},
+							{
+								Community: "10:101",
+								Prefixes:  []string{"192.168.1.0/24"},
+							},
+						}
+					}
+				},
+				validate: func(ppV4 []config.Peer, ppV6 []config.Peer, nodes []v1.Node) {
```

Comment:
can you add 
- a test where we have explicit prefixes, only one with community?
- a test where we have large communities
- a test where we have one address with multiple communities

Reply:
done, turns out we didn't have the code to support large communities, added a new commit (copy from MetalLB)